codegen/generator/nodejs/templates/src/api_test.go

```diff
@@ -0,0 +1,56 @@
+package test
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"flag"
+	"os"
+	"path/filepath"
+	"sort"
+	"testing"
+
+	"dagger.io/dagger"
+	"github.com/dagger/dagger/codegen/generator"
+	"github.com/stretchr/testify/require"
+)
+
+var update = flag.Bool("update", false, "update golden files")
+
+func TestAPI(t *testing.T) {
+	tmpl := templateHelper(t)
+
+	ctx := context.Background()
+	c, err := dagger.Connect(ctx)
+	require.NoError(t, err)
+
+	schema, err := generator.Introspect(ctx, c)
+	require.NoError(t, err)
+
+	generator.SetSchemaParents(schema)
+
+	sort.SliceStable(schema.Types, func(i, j int) bool {
+		return schema.Types[i].Name < schema.Types[j].Name
+	})
+	for _, v := range schema.Types {
+		sort.SliceStable(v.Fields, func(i, j int) bool {
+			return v.Fields[i].Name < v.Fields[j].Name
+		})
+	}
+
+	var b bytes.Buffer
+
+	err = tmpl.ExecuteTemplate(&b, "api", schema.Types)
+	require.NoError(t, err)
+
+	goldenPath := filepath.Join("testdata", "want-api-full.golden.ts")
```

Comment:
Won't this break every time we change the API, which is a couple of times a week?

Reply:
I mean, yes.
It was mostly for me, to help me navigate the creation of this big file without losing info by regression.
We could deactivate this, or, we could hide it behind some build tag to track in our CI, to make sure we do the correct semantic versioning update when our API breaks too much?
Also, this is the same file as the one in the nodejs sdk, so we could at least reduce to only 1 file.

Reply:
> We could deactivate this, or, we could hide it behind some build tag to track in our CI, to make sure we do the correct semantic versioning update when our API breaks too much?

Generation is an offline process. It's manually run and pushed as a PR when the API changes, so we already have good track of changes for all languages.

In `lint`, we're already checking every time that the committed code matches the code that's generated, so I feel it's redundant

Reply:
I mean, it was very useful for me during development. Now that it will be smaller API changes, I could remove it.
Worst case, the revert will still be there, and I can put it back with one command, at least locally.

Reply:
I removed it as it is covered by `mage sdk:nodejs:lint`.